libs/ha/ha_common_libs_k8s.py

```diff
@@ -1581,3 +1583,114 @@ def failover_pod(pod_obj, pod_list, failover_node):
                 LOGGER.error("Failed to failover pod %s to %s due to error: %s", pod,
                              failover_node, error)
                 return False, error
+
+    def mark_resource_failure(self, mnode_obj, pod_list: list, go_random: bool = True,
+                              rsc_opt: str = "mark_node_failure", rsc: str = "node",
+                              validate_set: bool = True):
+        """
+        Helper function to set resource status to Failed random if go_random.
+        :param pod_list: List of resource to be marked as failed
+        :param go_random: If True, send mark failure signal to resource randomly
+        :param mnode_obj: Master node object to fetch the resource ID
+        :param rsc_opt: Operation to be performed on resource (eg. mark_node_failure)
+        :param rsc: resource type (eg. node, cluster)
+        :param validate_set: If set to TRUE, its validate SET failure for resource
+        :return: bool, response
+        """
+        if rsc == 'node':
+            pod_info = {}
```

Comment:
If possible can we add logger that we are creating pod_info based on node/cluster?